src/lwc/rd2EntryForm/rd2EntryForm.js

```diff
@@ -211,13 +217,66 @@ export default class rd2EntryForm extends LightningElement {
     }
 
     /***
-    * @description Checks if credit card widget should be displayed
+    * @description Recurring Type change might hide or display the credit card widget
+    * @param event
+    */
+    handleRecurringTypeChange(event) {
+        this.handleElevateWidgetDisplay();
+    }
+
+    /***
+     * @description Currency change might hide or display the credit card widget
+     * @param event
+     */
+    handleCurrencyChange(event) {
+        this.handleElevateWidgetDisplay();
+    }
+
+    /***
+    * @description Checks if the credit card widget should be displayed.
+    */
+    handleElevateWidgetDisplay() {
+        if (this.isElevateCustomer) {
+            this.evaluateElevateWidget(this.getPaymentMethod());
+        }
+    }
+
+    /***
+    * @description Checks if the credit card widget should be displayed.
+    * The Elevate widget is applicable to new RDs only.
     * @param paymentMethod Payment method
     */
     evaluateElevateWidget(paymentMethod) {
         this.isElevateWidgetDisplayed = this.isElevateCustomer === true
-            && !this.isEdit // The Elevate widget is applicable to new RDs only
-            && paymentMethod === 'Credit Card';// Verify the payment method value
+            && !this.isEdit
+            && paymentMethod === PAYMENT_METHOD_CREDIT_CARD
+            && this.scheduleComponent.getRecurringType() === RECURRING_TYPE_OPEN
+            && this.isCurrencyUSD();
+    }
+
+    /***
+     * @description Returns value of the Payment Method field
+     */
+    getPaymentMethod() {
+        const paymentMethod = this.template.querySelector(`lightning-input-field[data-id='${FIELD_PAYMENT_METHOD.fieldApiName}']`);
+
+        return paymentMethod ? paymentMethod.value : null;
+    }
+
+    /***
+     * @description Returns true if the currency code on the Recurring Donatation is USD
+     */
+    isCurrencyUSD() {
+        let currencyCode;
+
+        if (this.isMultiCurrencyEnabled) {
+            currencyCode = this.template.querySelector('lightning-input-field[data-id="currencyField"]').value;
+
+        } else {
+            currencyCode = CURRENCY;
+        }
+
+        return currencyCode === CURRENCY_CODE_USD;
```

Comment:
I realize that it's not needed today, but do you think this should be more of a contains in list? Something like:
```
const ELEVATE_SUPPORTED_CURRENCIES = [ 'USD' ];
...
...
...
return ELEVATE_SUPPORTED_CURRENCIES.includes(currencyCode);
```

Reply:
Valid point, it has its benefits with regards to extending the functionality down the road. I need to rename all methods evaluating the currency then. But also, should we write the address validation code to be generic even though requirements are to allow US address and USD currency only?